### PR TITLE
Cart Page

### LINKED ISSUE #35: Cart Page

### PR DESCRIPTION
**Pull request by cuygur**
Cart container with show cart and single coupon components

- Desktop

cart-desktop

- Mobile

cart-mobile
#35

### CODE REVIEW

File: src/components/SingleCoupon/index.jsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                className="cardButton"
+                type="primary"
+                onClick={() => {
+                  if (amount > 1) {
+                    setAmount(amount - 1);
+                    changeAmount(coupon ? coupon.amount : null, "decrement");
+                  }
+                }}
+              >
+                -
```
Comment (halitbatur) [inline src/components/SingleCoupon/index.jsx:57]: extra line here

File: src/components/SingleCoupon/index.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                  deleteSingleCoupon(coupon.id);
+                }}
+              >
+                x
+              </Button>
+            )}
+            <Title>{coupon ? coupon.organizationName : null}</Title>
+            <img src={coupon ? coupon.img : null} alt="logo" width="120em" />
+            <Title>
+              {coupon ? coupon.amount : null}
```
Comment (halitbatur) [inline src/components/SingleCoupon/index.jsx:41]: why is this checking if coupon is there, I think this component doesn't get rendered at all if there was no coupon

File: src/components/ShowCart/index.jsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        </Col>
+      </Row>
+      <br />
+      <br />
+      <Row gutter={[48, 48]} justify="center" align="bottom">
+        <Col>
+          {isTesting ? (
+            <button type="button">{}</button>
+          ) : (
+            <Button
```
Comment (bertacsevercan) [inline src/components/ShowCart/index.jsx:71]: Can you change the style in these buttons. Use type="primary" like in antd and change the background of it maybe

File: src/components/SingleCoupon/index.jsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  return (
+    <div>
+      <Row>
+        <Col>
+          <Card hoverable align="center" className="couponCards">
+            {isTesting ? (
+              <button type="button">x</button>
+            ) : (
+              <Button
```
Comment (bertacsevercan) [inline src/components/SingleCoupon/index.jsx]: Can you put this button on the right, not on the middle ?

File: src/components/ShowCart/index.jsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            >
+              {t("cartCont.showCartComp.clearCart")}
+            </Button>
+          )}
+        </Col>
+        <Col>
+          {isTesting ? (
+            <button type="button">{}</button>
+          ) : (
+            <Button className="cartButton gradientCheckout">
```
Comment (bertacsevercan) [inline src/components/ShowCart/index.jsx]: Style of the button is a bit strange here as well

**Review by bertacsevercan — CHANGES_REQUESTED** [review]

File: src/components/ShowCart/index.jsx (hunk 3 above)
Comment (cuygur) [reply src/components/ShowCart/index.jsx:71]: I think it's nice

File: src/components/ShowCart/index.jsx (hunk 5 above)
Comment (cuygur) [reply src/components/ShowCart/index.jsx]: we can discuss this in the meeting

File: src/components/SingleCoupon/index.jsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                className="cardDeleteButton"
+                type="primary"
+                onClick={() => {
+                  changeAmount(coupon.amount * amount, "decrement");
+                  deleteSingleCoupon(coupon.id);
+                }}
+              >
+                x
+              </Button>
+            )}
```
Comment (ihab-sensei) [inline src/components/SingleCoupon/index.jsx]: Can you make the x a capital letter `x => X`

**Review by ihab-sensei — CHANGES_REQUESTED** [review]

File: src/components/SingleCoupon/index.jsx (hunk 4 above)
Comment (cuygur) [reply src/components/SingleCoupon/index.jsx]: in right, it's braking style, now it's on the left

File: src/components/ShowCart/index.jsx (hunk 3 above)
Comment (bertacsevercan) [reply src/components/ShowCart/index.jsx:71]: Yeah but it's not consistent with other buttons in the website

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/components/SingleCoupon/index.jsx (hunk 1 above)
Comment (cuygur) [reply src/components/SingleCoupon/index.jsx:57]: this - is for decrement

File: src/components/SingleCoupon/index.jsx (hunk 2 above)
Comment (cuygur) [reply src/components/SingleCoupon/index.jsx:41]: I wrote all like that to pass snapshot test

**Comment by halitbatur** [comment]
@cuygur resolve conflicts

**Comment by cuygur** [comment]
- new style

new style

**Review by halitbatur — APPROVED** [review]
lgtm

**Review by bertacsevercan — APPROVED** [review]
lgtm

**Pull request merged**